src/argilla/client/feedback/training/schemas.py

```diff
@@ -1064,13 +1130,208 @@ def _dict_to_format(ds: datasets.Dataset) -> List[Dict[str, List[Dict[str, str]]
+
+                self._test_output_formatting_func(output)
+
+                outputs.append(output)
+
+            if "label" in outputs[0]:
+                _all_labels = set()
+                for sample in outputs:
+                    if isinstance(sample, list):
+                        for response in sample:
+                            _all_labels.add(response["label"])
+                    else:
+                        _all_labels.add(sample["label"])
+
+                self.label = LabelQuestionUnification(
+                    question=LabelQuestion(name="custom_func", labels=list(_all_labels))
+                )
+
+            return outputs
+
+        else:
+            formatted_data = super()._format_data(dataset)
+            # NOTE: Maybe this post processing of the formatted data can be simplified
+            # or directly done in super()._format_data(dataset).
+            new_keys = {field.name: f"sentence-{i+1}" for i, field in enumerate(self.texts)}
+            if self.label:
+                new_keys.update({self.label.question.name: "label"})
+
+            outputs = []
+            for example in formatted_data:
+                record = {}
+                for k, v in new_keys.items():
+                    value = example[k]
+                    if v == "label":
+                        # At this point the label must be either an int or a float, determine which one is it.
+                        if value.lstrip("-").isdigit():
+                            value = int(value)
+                        else:
+                            value = float(value)
+                    record[v] = value
+                outputs.append(record)
+
+            return outputs
+
+    def unify_responses(self, responses: List[FeedbackRecord]):
+        self.label.strategy.unify_responses(responses=responses, field=self.label.question)
+
+    @requires_version("scikit-learn")
+    def _train_test_split(
+        self, data: List[dict], train_size: float, seed: int, stratify=None
+    ) -> Tuple[List[dict], List[dict]]:
+        from sklearn.model_selection import train_test_split
+
+        return train_test_split(data, train_size=train_size, shuffle=True, random_state=seed, stratify=stratify)
+
+    @requires_version("sentence-transformers")
+    def _prepare_for_training_with_sentence_transformers(
+        self, data: List[dict], train_size: float, seed: int
+    ) -> Union["InputExample", Tuple["InputExample", "InputExample"]]:
+        from sentence_transformers import InputExample
+
+        # Use the first sample to decide what type of dataset to generate:
+        sample_keys = data[0].keys()
+        if "label" in sample_keys:
+            if (len(sample_keys) == 3) and all(s in sample_keys for s in ["sentence-1", "sentence-2"]):
+                dataset_fields = lambda sample: {
+                    "texts": [sample["sentence-1"], sample["sentence-2"]],
+                    "label": sample["label"],
+                }
```

Comment:
My recommendation is to use set equality here, then we only need one if-elif-else tree. E.g.
```suggestion
        if set(sample_keys) == {"label", "sentence-1", "sentence-2"}:
            dataset_fields = lambda sample: {
                "texts": [sample["sentence-1"], sample["sentence-2"]],
                "label": sample["label"],
            }
```
And then the same for all cases here

Reply:
Done, far more readable like that :smile: